static/js/place/place_template.tsx

```diff
@@ -466,6 +466,150 @@ class ChildPlace extends Component<ChildPlacePropType, unknown> {
+   */
+  chartData: CachedStatVarDataMap;
+}
+
+class ChartBlock extends Component<ChartBlockPropType, unknown> {
+  constructor(props: ChartBlockPropType) {
+    super(props);
+  }
+
+  render() {
+    const configList = this.props.isOverview
+      ? this.buildOverviewConfig(this.props.placeType, this.props.config)
+      : this.buildTopicConfig(this.props.placeType, this.props.config);
+    return (
+      <div className="chart-block">
+        <div className="chart-block-title">{this.props.config.title}</div>
+        <div className="row row-cols-md-2 row-cols-1">
+          {configList.map((item) => {
+            const id = randDomId();
+            return (
+              <Chart
+                key={id}
+                id={id}
+                config={item}
+                dcid={this.props.dcid}
+                placeType={this.props.placeType}
+                parentPlaces={this.props.parentPlaces}
+                childPlaces={this.props.childPlaces}
+                similarPlaces={this.props.similarPlaces}
+                nearbyPlaces={this.props.nearbyPlaces}
+                chartData={this.props.chartData}
+              />
+            );
+          })}
+        </div>
+      </div>
+    );
+  }
+
+  private buildOverviewConfig(placeType: string, config: ConfigType) {
+    const result = [];
+    let conf = { ...config };
+    conf.chartType = chartTypeEnum.LINE;
+    conf.title = "TREND";
+    result.push(conf);
+
+    conf = { ...config };
+    conf.chartType = chartTypeEnum.GROUP_BAR;
+    conf.axis = "PLACE";
+    if (placeType == "Country") {
+      // Containing place chart
+      conf.title = "Containing Places";
+      conf.placeRelation = placeRelationEnum.CONTAINING;
+    } else {
+      conf.title = "Nearby Places";
+      conf.placeRelation = placeRelationEnum.NEARBY;
+    }
+    result.push(conf);
+    return result;
+  }
+
+  private buildTopicConfig(placeType: string, config: ConfigType) {
+    const result: ConfigType[] = [];
+    let conf = { ...config };
+    conf.chartType = chartTypeEnum.LINE;
+    conf.title = "TREND";
+    result.push(conf);
+
+    if (placeType != "Country") {
+      // Nearby places
+      conf = { ...config };
+      conf.chartType = chartTypeEnum.GROUP_BAR;
+      conf.placeRelation = placeRelationEnum.NEARBY;
+      conf.axis = "PLACE";
+      conf.title = "Nearby Places";
+      result.push(conf);
+      // Similar places
+      conf = { ...config };
+      conf.chartType = chartTypeEnum.GROUP_BAR;
+      conf.placeRelation = placeRelationEnum.SIMILAR;
+      conf.axis = "PLACE";
+      conf.title = "Similar Places";
+      result.push(conf);
+    }
+    if (placeType != "City") {
+      // Children places
+      conf = { ...config };
+      conf.chartType = chartTypeEnum.GROUP_BAR;
+      conf.placeRelation = placeRelationEnum.CONTAINING;
+      conf.axis = "PLACE";
+      conf.title = "Children Places";
+      result.push(conf);
+    } else {
```

Comment:
maybe skip this for state too (there'll only be one parent place to compare)

Reply:
sg, State is handled in the if condition